team/bundles/org.eclipse.compare/compare/org/eclipse/compare/contentmergeviewer/TextMergeViewer.java

```diff
@@ -4232,6 +4232,16 @@ private boolean isAnySideEditable() {
 		return isLeftEditable() || isRightEditable();
 	}
 
+	/**
+	 * The line height does not only depend on the line height itself, but also on
+	 * the line spacing etc.
+	 */
+	private int getRendererLineHeight(MergeSourceViewer tp) {
+		return tp.getSourceViewer().getTextWidget().getLineHeight()
```

Comment:
1) Could we avoid duplication and replace this with 

```
	private int getRendererLineHeight(MergeSourceViewer tp) {
		StyledText text = tp.getSourceViewer().getTextWidget();
		return text.getLineHeight() + text.getLineSpacing() + text.getLineVerticalIndent(0);
	}
```
2) `getLineVerticalIndent(0)` seem to be hard-coded to the first line. Do we assume it never changes?

Reply:
1. Sure.
2. I think this is actually a requirement with the StyledTextRenderer. 

In the class StyledTextRenderer in the method: getLineHeight(int lineIndex, boolean exact)
calculate the line height for a specific line will be calculated with:
getLineHeight() + getLineSpacing(lineIndex) + getLineVerticalIndent(lineIndex);

And in the paint methods where this method will be used, that what is actually wanted is not a line height, but the height from all lines before:
see for example Line 4438 in this file: 
	int y= (region.x * lineHeight) + shift;

here y is wanted, region.x are the number of lines.

I am currently not happy with this change, since you are right, each line could have different vertical indent in theory.

Reply:
1. Done
2. I created a new method, which sums up for each line the indent and the spacing. This leads to an exact result.